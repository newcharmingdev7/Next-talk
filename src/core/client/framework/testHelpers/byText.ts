import React from "react";
import { ReactTestInstance } from "react-test-renderer";

import matchText, { TextMatchOptions, TextMatchPattern } from "./matchText";

const matcher = (pattern: TextMatchPattern, options?: TextMatchOptions) => (
  i: ReactTestInstance
) => {
  // Only look at dom components.
  if (typeof i.type !== "string") {
    return false;
  }
  if (!i.props.children) {
    return false;
  }
  const children = React.Children.toArray(i.props.children);
  for (const c of children) {
    if (typeof c === "string" && matchText(pattern, c, options)) {
      return true;
    }
  }
  return false;
};

export function getByText(
  container: ReactTestInstance,
  pattern: TextMatchPattern,
  options?: TextMatchOptions
) {
  return container.find(matcher(pattern, options));
}

export function getAllByText(
  container: ReactTestInstance,
  pattern: TextMatchPattern,
  options?: TextMatchOptions
) {
  const results = container.findAll(matcher(pattern, options));
  if (!results.length) {
    throw new Error(`Couldn't find text ${pattern}`);
  }
  return results;
}

export function queryByText(
  container: ReactTestInstance,
  pattern: TextMatchPattern,
  options?: TextMatchOptions
) {
  const results = container.findAll(matcher(pattern, options));
  if (!results.length) {
    return null;
  }
  return results[0];
}

export function queryAllByText(
  container: ReactTestInstance,
  pattern: TextMatchPattern,
  options?: TextMatchOptions
) {
  return container.findAll(matcher(pattern, options));
}
