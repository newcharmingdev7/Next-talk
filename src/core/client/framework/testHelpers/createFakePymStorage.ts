import { PymStorage } from "talk-framework/lib/storage";

export class FakeStorage implements PymStorage {
  public store: Record<string, string> = {};

  public setItem(key: string, value: string) {
    this.store[key] = value;
    return Promise.resolve();
  }
  public removeItem(key: string) {
    delete this.store[key];
    return Promise.resolve();
  }
  public getItem(key: string) {
    return Promise.resolve(this.store[key] || null);
  }
}

export default function createFakePymStorage() {
  return new FakeStorage();
}
