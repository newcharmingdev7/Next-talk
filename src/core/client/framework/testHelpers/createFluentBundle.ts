import { FluentBundle } from "fluent/compat";
import fs from "fs";
import path from "path";

// These locale prefixes are always loaded.
const commonPrefixes = ["common", "framework"];

function decorateErrorWhenMissing(bundle: FluentBundle) {
  const originalHasMessage = bundle.hasMessage;
  const originalGetMessage = bundle.getMessage;
  const missing: string[] = [];
  bundle.hasMessage = (id: string) => {
    const result = originalHasMessage.apply(bundle, [id]);
    if (!result) {
      const msg = `${bundle.locales} translation for key "${id}" not found`;
      // tslint:disable-next-line:no-console
      console.error(msg);
      missing.push(id);
    }
    // Even if it is missing, we say it is available and later return a descriptive error
    // string as the translation.
    return true;
  };
  bundle.getMessage = (id: string) => {
    if (missing.indexOf(id) !== -1) {
      return `Missing translation "${id}"`;
    }
    return originalGetMessage.apply(bundle, [id]);
  };
  return bundle;
}

function createFluentBundle(
  target: string,
  pathToLocale: string
): FluentBundle {
  const bundle = new FluentBundle("en-US");
  const files = fs.readdirSync(pathToLocale);
  const prefixes = commonPrefixes.concat(target);
  files.forEach(f => {
    prefixes.forEach(prefix => {
      if (f.startsWith(prefix)) {
        bundle.addMessages(require(path.resolve(pathToLocale, f)));
      }
    });
  });
  return decorateErrorWhenMissing(bundle);
}

export default createFluentBundle;
