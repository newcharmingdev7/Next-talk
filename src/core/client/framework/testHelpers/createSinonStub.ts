import sinon, { SinonStub } from "sinon";

export default function createSinonStub(
  ...callbacks: Array<(s: SinonStub) => void>
): SinonStub {
  const stub = sinon.stub();
  callbacks.forEach(cb => cb(stub));
  return stub;
}
