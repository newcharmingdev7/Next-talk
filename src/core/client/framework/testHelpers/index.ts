export {
  default as createRelayEnvironment,
  CreateRelayEnvironmentParams,
} from "./createRelayEnvironment";
export { default as createFluentBundle } from "./createFluentBundle";
export {
  default as removeFragmentRefs,
  NoFragmentRefs,
} from "./removeFragmentRefs";
