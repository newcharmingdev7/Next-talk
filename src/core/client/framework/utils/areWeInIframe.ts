/**
 * Returns true if we are in an iframe.
 */
export default function areWeInIframe() {
  try {
    // eslint-disable-next-line no-restricted-globals
    return window.self !== window.top;
  } catch (e) {
    return true;
  }
}
