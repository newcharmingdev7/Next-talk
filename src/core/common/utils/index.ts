export { default as timeout } from "./timeout";
export { default as pascalCase } from "./pascalCase";
