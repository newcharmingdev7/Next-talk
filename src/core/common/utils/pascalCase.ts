import { camelCase } from "lodash";

export default function pascalCase(value: string) {
  return value[0].toUpperCase() + camelCase(value).slice(1);
}
