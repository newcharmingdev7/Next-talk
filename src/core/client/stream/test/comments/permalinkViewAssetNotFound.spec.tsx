import { ReactTestRenderer } from "react-test-renderer";

import { timeout } from "talk-common/utils";
import { createSinonStub } from "talk-framework/testHelpers";

import { settings } from "../fixtures";
import create from "./create";

let testRenderer: ReactTestRenderer;
beforeEach(() => {
  const resolvers = {
    Query: {
      comment: () => null,
      asset: () => null,
      settings: createSinonStub(
        s => s.throws(),
        s => s.withArgs(undefined).returns(settings)
      ),
    },
  };

  ({ testRenderer } = create({
    // Set this to true, to see graphql responses.
    logNetwork: false,
    resolvers,
    initLocalState: localRecord => {
      localRecord.setValue("unknown-asset-id", "assetID");
      localRecord.setValue("unknown-comment-id", "commentID");
    },
  }));
});

it("renders permalink view with unknown asset", async () => {
  // Wait for loading.
  await timeout();
  expect(testRenderer.toJSON()).toMatchSnapshot();
});
