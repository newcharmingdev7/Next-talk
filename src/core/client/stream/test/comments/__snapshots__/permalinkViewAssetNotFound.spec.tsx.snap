// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders permalink view with unknown asset 1`] = `
<div
  className="HorizontalGutter-root App-root HorizontalGutter-full"
>
  <ul
    className="TabBar-root TabBar-primary"
    role="tablist"
  >
    <li
      className="Tab-root"
      id="tab-COMMENTS"
      role="presentation"
    >
      <button
        aria-controls="tabPane-COMMENTS"
        aria-selected={true}
        className="BaseButton-root Tab-button Tab-primary Tab-active"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        role="tab"
        type="button"
      >
        Comments
      </button>
    </li>
  </ul>
  <section
    aria-labelledby="tab-COMMENTS"
    id="tabPane-COMMENTS"
    role="tabpanel"
  >
    <div>
      Asset not found
    </div>
  </section>
</div>
`;
